src/binary/ibc-setup/commands/init.ts

```diff
@@ -57,6 +97,71 @@ export async function run(options: Options) {
   }
 
   const registry = yaml.load(fs.readFileSync(registryFilePath, 'utf-8'));
-  // TODO #75: registry file validation
-  console.log(registry);
+
+  const ajv = new Ajv({ allErrors: true });
+  const schema: JSONSchemaType<Registry> = {
+    type: 'object',
+    required: ['chains', 'version'],
+    additionalProperties: false,
+    properties: {
+      version: {
+        type: 'number',
+      },
+      chains: {
+        type: 'object',
+        minProperties: 2,
+        required: [options.src, options.dest], // ensure registry compatibility with "src" and "dest"
+        additionalProperties: false,
+        patternProperties: {
+          '^(.*)$': {
+            type: 'object',
+            required: ['chain_id', 'gas_price', 'hd_path', 'prefix', 'rpc'],
+            additionalProperties: false,
+            properties: {
+              chain_id: { type: 'string' },
+              prefix: { type: 'string' },
+              gas_price: { type: 'string' },
+              hd_path: { type: 'string' },
+              rpc: { type: 'array', items: { type: 'string' } },
+            },
+          },
+        },
+      },
+    },
+  };
+  const validate = ajv.compile(schema);
+  if (!validate(registry)) {
+    const errors = (validate.errors ?? []).map(
+      ({ dataPath, message }) => `"${dataPath}" ${message}`
+    );
+    throw new Error(
+      [`${registryFile} validation failed.`, ...errors].join(os.EOL)
+    );
```

Comment:
It produces this kind of errors:

```bash
Error: registry.yaml validation failed.
"/chains/local_wasm" should have required property 'chain_id'
```

Reply:
Yeah, looks nice.
When we add a logger, we can look if there is a better format, but it looks nice (and nice use of join)

Reply:
Of course, I think we would have to revisit all the error messages sooner or later. It'd be nice to have them in the same style and format.